### PR TITLE
prov/efa, fabtests/efa: Only test inter_min_read_write_size for for emulated write path 

### PR DESCRIPTION
**Pull request by shijin-aws**
FI_EFA_INTER_MIN_READ_WRITE_SIZE is only applied to emulated write protocols. When efa device support rdma write,
rdma write will be always used.
This PR introduces a efa_rdma_checker.c code to check efa device's rdma read/write capability on a given platform, and
skip inter_min_read_write_size test in fabtests if the device support rdma-write.

### CODE REVIEW

File: fabtests/prov/efa/src/efa_rdma_checker.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	struct efadv_device_attr efadv_attr = {0};
+	int dev_cnt;
+	int err, opt;
+	enum rdma_op op = READ;
+
+	while ((opt = getopt(argc, argv, "ho:")) != -1) {
+		switch (opt) {
+		case 'o':
+			if (!strcasecmp(optarg, "read")) {
+                op = READ;
```
Comment (jiaxiyan) [inline fabtests/prov/efa/src/efa_rdma_checker.c:38]: indent

Comment (shijin-aws) [reply fabtests/prov/efa/src/efa_rdma_checker.c]: fixed

**Comment by shijin-aws** [comment]
AWS CI failure is real

```
prov/efa/src/efa_rdma_checker.o: In function `main':

/home/ec2-user/PortaFiducia/build/libraries/libfabric/pr10411-dso/source/libfabric/fabtests/prov/efa/src/efa_rdma_checker.c:71: undefined reference to `efadv_query_device'

collect2: error: ld returned 1 exit status

make[1]: *** [prov/efa/src/fi_efa_rdma_checker] Error 1

```
Need to fix

**Comment by shijin-aws** [comment]
@j-xiong could u review https://github.com/ofiwg/libfabric/pull/10411/commits/69011622ba076d73d43352698417f127bb26a3a3 ?

File: fabtests/prov/efa/Makefile.include [hunk 2]
```diff
@@ -65,4 +68,11 @@ prov_efa_src_fi_efa_exhaust_mr_reg_rdm_pingpong_SOURCES = \
 	$(efa_exhaust_mr_reg_srcs) \
 	$(benchmarks_srcs)
 prov_efa_src_fi_efa_exhaust_mr_reg_rdm_pingpong_LDADD = libfabtests.la
+
+if HAVE_EFADV
+prov_efa_src_fi_efa_rdma_checker_SOURCES = \
+	prov/efa/src/efa_rdma_checker.c
+prov_efa_src_fi_efa_rdma_checker_LDADD = libfabtests.la
```
Comment (wenduwan) [inline fabtests/prov/efa/Makefile.include:75]: It's dubious to me whether `LDADD` is needed.

Reading https://www.gnu.org/software/automake/manual/html_node/Linking.html I think we can do `prov_efa_src_fi_efa_rdma_checker_LDFLAGS = -lefa` and get rid of `configure.m4`

Comment (shijin-aws) [reply fabtests/prov/efa/Makefile.include:72]: Good call out, It works for me. I added LDFLAGS and dropped https://github.com/ofiwg/libfabric/commit/69011622ba076d73d43352698417f127bb26a3a3.

**Comment by shijin-aws** [comment]
> @j-xiong could u review [6901162](https://github.com/ofiwg/libfabric/commit/69011622ba076d73d43352698417f127bb26a3a3) ?

I dropped this commit, it should be a pure efa change now.

**Review by wenduwan — APPROVED** [review]

**Comment by j-xiong** [comment]
@shijin-aws While I was reviewing that one the power went out. Good to see it has been dropped.

**Pull request merged**